### PR TITLE
ci: new cicd system

### PR DESCRIPTION
**Pull request by xavidop**
**Fixes or implements VF-2657**

### Brief description. What is this change?

Implements [this](https://www.notion.so/voiceflow/CI-CD-System-f80ef4dd66b8407ba23f7a72e097b96d)

### Checklist

- [ ] this is a breaking change and should publish a new major version
- [ ] appropriate tests have been written

### CODE REVIEW

File: orb.yml [hunk 1]
```diff
@@ -2217,10 +3077,44 @@ jobs:
...
-          published_tags: MONOREPO_UPDATED_TAGS
+      - when:
+          condition: << parameters.trigger_tags_pipelines >>
+          steps:
+            - trigger_tags_pipelines:
+                published_tags: MONOREPO_UPDATED_TAGS
+
+#-------------------------------------------------------------CDK JOBS----------------------------------------------------------
+
+  cdk-node-deploy:
```
Comment (frankgu968) [inline orb.yml:3090]: can we merge this with the go deploy step? they have pretty much identical functions. Duplicating this is not a great approach imo

**Review by frankgu968 — CHANGES_REQUESTED** [review]

File: orb.yml [hunk 2]
```diff
@@ -84,7 +84,7 @@ executors:
   go-executor:
     resource_class: medium+
     docker:
-      - image: cimg/go:1.17
+      - image: cimg/go:1.17-node
```
[the remark this answers is not in the document]
Comment (xavidop) [reply orb.yml:87]: because of cdk

File: orb.yml (hunk 1 above)
Comment (xavidop) [reply orb.yml]: yes! done!

File: orb.yml [hunk 3]
```diff
@@ -1877,6 +2404,32 @@ commands:
           run_on_root: << parameters.run_on_root >>
           force_execution: << parameters.force_execution >>
 
+  analyze_dependencies_monorepo:
```
[the remark this answers is not in the document]
Comment (xavidop) [reply orb.yml]: yes! done!

File: orb.yml [hunk 4]
```diff
@@ -321,6 +295,64 @@ commands:
...
+      check_commit_message:
+        description: The SSH key with write permissions to the repository
+        type: string
+        default: ""
+    steps:
+      - when:
+          condition: << parameters.checkout >>
+          steps:
+            - checkout
+      - add_ssh_keys: # To enable write access to repository for removing development environment branches
```
[the remark this answers is not in the document]
Comment (xavidop) [reply orb.yml]: yes! done!

**Review by frankgu968 — APPROVED** [review]

**Pull request merged**